.github/workflows/deploy.sh

```diff
@@ -25,47 +15,69 @@ printf "${RED}4. ACR Purge 🗑️${NC}\n\n"
 
 read selection
 
-if [ "$selection" = "0" ]
-then
-############### INFRA ###############
-destination=infrastructure
-############### INFRA ###############
-elif [ "$selection" = "1" ]
-then
-############### DEV ###############
-destination=dev
-############### DEV ###############
-elif [ "$selection" = "2" ]
-then
-############### STAGING ###############
-destination=staging
-############### STAGING ###############
-elif [ "$selection" = "3" ]
-then
-############### PRODUCTION ###############
-destination=prod
-############### PRODUCTION ###############
-elif [ "$selection" = "4" ]
-then
-############### ACR PURGE ###############
-destination=""
-az acr run --cmd "acr purge --filter 'portal-ui:.*' --filter 'gef-ui:.*' --filter 'trade-finance-manager-ui:.*' --filter 'portal-api:.*' --filter 'trade-finance-manager-api:.*' --filter 'dtfs-central-api:.*' --filter 'reference-data-proxy:.*' --filter 'azure-function-number-generator:.*' --filter 'azure-function-acbs:.*' --ago 15d" --registry tfsdev /dev/null
-############### ACR PURGE ###############
-fi
+if [ -n "$selection" ]; then
+
+    if [ "$selection" = "0" ]
+    then
+    ############### INFRA ###############
+    destination=infrastructure
+    branch=main
+    ############### INFRA ###############
+    elif [ "$selection" = "1" ]
+    then
+    ############### DEV ###############
+    destination=dev
+    branch=main-amendments
+    ############### DEV ###############
+    elif [ "$selection" = "2" ]
+    then
+    ############### STAGING ###############
+    destination=staging
+    branch=main
+    ############### STAGING ###############
+    elif [ "$selection" = "3" ]
+    then
+    ############### PRODUCTION ###############
+    destination=prod
+    branch=main
+    ############### PRODUCTION ###############
+    elif [ "$selection" = "4" ]
+    then
+    ############### ACR PURGE ###############
+    destination=""
+    branch=""
+    az acr run --cmd "acr purge --filter 'portal-ui:.*' --filter 'gef-ui:.*' --filter 'trade-finance-manager-ui:.*' --filter 'portal-api:.*' --filter 'trade-finance-manager-api:.*' --filter 'dtfs-central-api:.*' --filter 'reference-data-proxy:.*' --filter 'azure-function-number-generator:.*' --filter 'azure-function-acbs:.*' --ago 15d" --registry tfsdev /dev/null
+    az acr run --cmd "acr purge --filter 'portal-ui:.*' --filter 'gef-ui:.*' --filter 'trade-finance-manager-ui:.*' --filter 'portal-api:.*' --filter 'trade-finance-manager-api:.*' --filter 'dtfs-central-api:.*' --filter 'reference-data-proxy:.*' --filter 'azure-function-number-generator:.*' --filter 'azure-function-acbs:.*' --ago 15d" --registry tfsstaging /dev/null
+    ############### ACR PURGE ###############
+    fi
+
+    if [ -n "$destination" -a -n "$branch" ]
+    then
+    # Display latest push commit
+    git checkout ${branch}
```

Comment:
do we need to use:
git checkout `"${branch}"`  ? 
we are getting some linting error on it.